packages/create-cloudflare/src/__tests__/workers.integration.test.ts

```diff
@@ -0,0 +1,119 @@
+import { mkdtempSync, rmSync } from "fs";
+import { tmpdir } from "os";
+import { join } from "path";
+import { getWorkerdCompatibilityDate } from "helpers/command";
+import { readFile, writeFile } from "helpers/files";
+import { describe, test, afterEach, beforeEach, expect } from "vitest";
+import * as workers from "../workers";
+import { createTestContext } from "./helpers";
+import type { C3Context } from "types";
+
+describe("updateWranglerToml", () => {
+	let ctx: C3Context;
+	let tmpDirPath: string;
+	let wranglerTomlPath: string;
+
+	beforeEach(() => {
+		ctx = createTestContext();
+		tmpDirPath = mkdtempSync(join(tmpdir(), `c3-tests-updateWranglerToml`));
+		wranglerTomlPath = join(tmpDirPath, "wrangler.toml");
+		ctx.project.path = tmpDirPath;
+	});
+
+	afterEach(() => {
+		rmSync(tmpDirPath, { recursive: true, force: true });
+	});
+
+	test("placeholder replacement", async () => {
+		const toml = `
+name = "<TBD>"
+main = "src/index.ts"
+compatibility_date = "<TBD>"
+	  `;
+
+		writeFile(wranglerTomlPath, toml);
+		await workers.updateWranglerToml(ctx);
+
+		const newToml = readFile(wranglerTomlPath);
+		expect(newToml).toMatch(`name = "${ctx.project.name}"`);
+		expect(newToml).toMatch(`main = "src/index.ts"`);
+		expect(newToml).toMatch(
+			`compatibility_date = "${await getWorkerdCompatibilityDate()}"`
+		);
+	});
+
+	test("empty replacement", async () => {
+		const toml = `
+name =
+main = "src/index.js"
+compatibility_date =
+    `;
+
+		writeFile(wranglerTomlPath, toml);
+		await workers.updateWranglerToml(ctx);
+
+		const newToml = readFile(wranglerTomlPath);
+		expect(newToml).toMatch(`name = "${ctx.project.name}"`);
+		expect(newToml).toMatch(`main = "src/index.js"`);
+		expect(newToml).toMatch(
+			`compatibility_date = "${await getWorkerdCompatibilityDate()}"`
+		);
+	});
+
+	test("empty string replacement", async () => {
+		const toml = `
+name = "" # todo
+main = "src/index.js"
+    `;
+
+		writeFile(wranglerTomlPath, toml);
+		await workers.updateWranglerToml(ctx);
+
+		const newToml = readFile(wranglerTomlPath);
+		expect(newToml).toMatch(`name = "${ctx.project.name}"`);
+	});
+
+	test("string literal replacement", async () => {
+		const toml = `
+name = "my-cool-worker"
+main = "src/index.ts"
+    `;
+
+		writeFile(wranglerTomlPath, toml);
+		await workers.updateWranglerToml(ctx);
+
+		const newToml = readFile(wranglerTomlPath);
+		expect(newToml).toMatch(`name = "${ctx.project.name}"`);
+	});
+
+	test("replace old compatibility date", async () => {
```

Comment:
I am not sure about this.... if the toml contains a specific date, could there be a reason for it? wouldn't it be safer/clearer to be explicit and always rely on `<TBD>` instead of potentially introduce surprising behaviors?

Reply:
There could be a reason for it, but in some cases the template could just be stale (like  [this one](https://github.com/cloudflare/workers-sdk/blob/main/templates/worker/wrangler.toml) or [this one](https://github.com/cloudflare/workers-sdk/blob/main/templates/worker-typescript/wrangler.toml). It looks most of our existing template are somewhat stale and don't have an automated process for staying up to date.

The original idea behind this particular change was two fold: 1) to allow a low-friction migration of the existing templates, and 2) to maintain a consistent handling of templates whether remote or internal. 

Regarding 1, If we were to rely on `<TBD>` here we'd need to update all the existing templates to use this placeholder explicitly. But when we do that, users won't be able to simply `git clone` them like they do today, since they'll need to look up an actual compat date and put it in their `wrangler.toml`. 

Regarding 2, I think even if we respect existing compatibility dates in remote templates, we want to force always using latest for internal templates. We'll want to keep the internal templates up to date at all times, and if compat issues arise we'll want to see them fail in our tests so we can update them. Without this, if an internal template author mistakenly adds an actual compat date instead of a placeholder, the template would go stale over time without us realizing it.

I think you're right, it is safer only replace an explicit placeholder. But I think that means we should completely avoid using `<TBD>` placeholders in our library of external templates to avoid adding friction to people who still want `git clone` the templates instead of running them with C3. To solve the staleness issue we'll just need to come up with some automation to keep them up to date, and encourage template authors to do the same.

Reply:
I see 🙂 

Thanks a lot for the explanation 🙏 

I'm ok with how it works then, although I must admit that I am not completely sold, but if you strongly believe that this is the way to go I am fine with it 🙂

___

Basically with the current logic of updating the date, it's going to be impossible for template authors to actually specify a precise compat date for their worker, not having this possibility makes me a bit uneasy, but I also imagine such use case not to be common (if ever needed at all!) so I guess it should be fine 🙂